### PR TITLE
add docs about building from source

### PR DESCRIPTION
**Pull request by gy95**
* **Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
add docs about building from source

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

File: content/en/docs/setup/local.md [hunk 1]
```diff
@@ -87,3 +87,20 @@ Start edgecore:
 ```
 
 Run `edgecore -h` to get help info and add options if needed.
+
+### Build from source
+If you want to build KubeEdge from source, there are two options:
+
+You have a working Go environment.
```
Comment (kevin-wangzefeng) [inline content/en/docs/setup/local.md:94]:
```suggestion
You have a working Go environment, and would like to build on host directly.
```

File: content/en/docs/setup/local.md [hunk 2]
```diff
@@ -87,3 +87,20 @@ Start edgecore:
...
+```shell
+git clone https://github.com/kubeedge/kubeedge.git
+cd kubeedge
+make BUILD_WITH_CONTAINER=false
+```
+
+You have a working Docker environment.
```
Comment (kevin-wangzefeng) [inline content/en/docs/setup/local.md:101]:
```suggestion
You have a working Docker environment. And enjoin building inside container for simpilified environment consistency.
```

**Review by Arhell — COMMENTED** [review]
/lgtm

File: content/en/docs/setup/local.md [hunk 3]
```diff
@@ -87,3 +87,20 @@ Start edgecore:
...
+If you want to build KubeEdge from source, there are two options:
+
+You have a working Go environment, and would like to build on host directly.
+```shell
+git clone https://github.com/kubeedge/kubeedge.git
+cd kubeedge
+make BUILD_WITH_CONTAINER=false
+```
+
+You have a working Docker environment, and enjoy building inside container for simpilified environment consistency.
```
Comment (anvithks) [inline content/en/docs/setup/local.md:101]: Typo: simpilified -> simplified

**Review by anvithks — CHANGES_REQUESTED** [review]
Please fix typo. @kevin-wangzefeng please check for the correctness of the commands

**Pull request closed without merging**